client/tufclient.go

```diff
@@ -78,25 +79,81 @@ func (c *TUFClient) update() error {
 	return nil
 }
 
-// downloadRoot is responsible for downloading the root.json
-func (c *TUFClient) downloadRoot() error {
+// updateRoot checks if there is a newer version of the root available, and if so
+// downloads all intermediate root files to allow proper key rotation.
+func (c *TUFClient) updateRoot() error {
 	role := data.CanonicalRootRole
-	consistentInfo := c.newBuilder.GetConsistentInfo(role)
 
-	// We can't read an exact size for the root metadata without risking getting stuck in the TUF update cycle
-	// since it's possible that downloading timestamp/snapshot metadata may fail due to a signature mismatch
-	if !consistentInfo.ChecksumKnown() {
-		logrus.Debugf("Loading root with no expected checksum")
+	// Get Current Root Version
```

Comment:
The old builder should already have the root role loaded (from `NotaryRepository.bootstrapClient`), if the one on disk was valid.  If a new one has been downloaded since, it's only because it was validated against the previous one, so no updates should be necessary.  So I think we can just call `c.oldBuilder.GetLoadedVersion(role)` here?